### ISSUE TITLE
Set usec timeout for connection

### ISSUE THREAD

**Issue by deStrO** [koraktor/steam-condenser#88]
Hello,

When we do a rcon command, we have to wait 1 second to see the result, in LAN or on low ping server, a usec timeout is better. 

public function select($timeout = 0, $stimeout = 0)
    {
        $read = array($this->socket);
        $write = null;
        $except = null;

```
    if($this->socketsEnabled)
    {
        $select = socket_select($read, $write, $except, $timeout, $stimeout);
    }
    else
    {
        $select = stream_select($read, $write, $except, $timeout, $stimeout);
    }

    return $select > 0;
}
```

define('STEAM_CONDENSER_DEFAULT_TIMEOUT',0);
define('STEAM_CONDENSER_DEFAULT_TIMEOUT_USEC',1000);

And in rconsocket at line 59:
if(!$this->channel->socket()->select(STEAM_CONDENSER_DEFAULT_TIMEOUT,STEAM_CONDENSER_DEFAULT_TIMEOUT_USEC))

**Comment by koraktor** [comment]
Sounds like a valuable addition. I was always unsatisfied by the current implementation. But it won't get a lot better. The RCON isn't very smart.

I'll try to find equal solutions for Ruby and Java and integrate them too.

**Comment by deStrO** [comment]
And in TCPSocket.php

at line 61:
throw new Exception("Could not create socket: $socketErrstr");

I catched the error without know the reason, it's more practice if you know the error :D

**Comment by koraktor** [comment]
Allow customization of socket timeouts

Timeouts are now based on milliseconds in all three implementations.
Default is 1000ms (1s).

Closed by 4c245ccc93ec6b3296f3b08204b65bb7db6d5292

**Comment by koraktor** [comment]
As you may see, I took a bit different approach also suitable for easier customization of the timeout setting.

 The "Could not create socket" error should also be more verbose now.
